evap/staff/views.py

```diff
@@ -644,21 +644,46 @@ def questionnaire_create(request):
         return render(request, "staff_questionnaire_form.html", dict(form=form, formset=formset))
 
 
+def make_questionnaire_edit_forms(request, questionnaire, editable):
+    if editable:
+        InlineQuestionFormset = inlineformset_factory(Questionnaire, Question, formset=AtLeastOneFormSet, form=QuestionForm, extra=1, exclude=('questionnaire',))
+    else:
+        question_count = questionnaire.question_set.count()
+        InlineQuestionFormset = inlineformset_factory(Questionnaire, Question, formset=AtLeastOneFormSet, form=QuestionForm, extra=0, exclude=('questionnaire',),
+                                                      can_delete=False, max_num=question_count, validate_max=True, min_num=question_count, validate_min=True)
+
+    form = QuestionnaireForm(request.POST or None, instance=questionnaire)
+    formset = InlineQuestionFormset(request.POST or None, instance=questionnaire)
+
+
+    if not editable:
+        editable_fields =  ['staff_only', 'obsolete', 'name_de', 'name_en', 'description_de', 'description_en']
+        for name, field in form.fields.items():
+            print(name)
+            if name not in editable_fields:
+                field.disabled = True
+        for question_form in formset.forms:
+            for name, field in question_form.fields.items():
+                if name is not 'id':
+                    field.disabled = True
+
+    return form, formset
+
+
 @staff_required
 def questionnaire_edit(request, questionnaire_id):
     questionnaire = get_object_or_404(Questionnaire, id=questionnaire_id)
-    InlineQuestionFormset = inlineformset_factory(Questionnaire, Question, formset=AtLeastOneFormSet, form=QuestionForm, extra=1, exclude=('questionnaire',))
+    editable = questionnaire.can_staff_edit
 
-    form = QuestionnaireForm(request.POST or None, instance=questionnaire)
-    formset = InlineQuestionFormset(request.POST or None, instance=questionnaire)
+    if not editable:
+        messages.info(request, _("Some fields are disabled as this questionnaire is already in use."))
```

Comment:
calling this here shows the message also when the form was valid and the user was redirected to the index page after saving changes.